File: java/src/main/java/com/lacombe/promo3/taxi/Taxi.java
package com.lacombe.promo3.taxi;

public class Taxi {

    private String name;

    public Taxi(String name) {

        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Taxi taxi = (Taxi) o;

        return name != null ? name.equals(taxi.name) : taxi.name == null;
    }

    @Override
    public int hashCode() {
        return name != null ? name.hashCode() : 0;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: dotnet/Socrates/TaxiOrganization/Taxi.cs
using System.Collections.Generic;

namespace Socrates.TaxiOrganization
{
    public class Taxi
    {
        private readonly string name;

        public Taxi(string name)
        {
            this.name = name;
        }

        public override bool Equals(object obj)
        {
            var taxi = obj as Taxi;
            return taxi != null &&
                   name == taxi.name;
        }

        public override int GetHashCode()
        {
            return 363513814 + 
                EqualityComparer<string>.Default.GetHashCode(name);
        }

        public override string ToString()
        {
            return name;
        }
    }
}
